### PR TITLE
[TBBAS-1808] .NET Bindings - Fill in setting-up guide for .NET in "Getting started"

### PR DESCRIPTION
**Pull request by Daniel-BB**
# Type of change:

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# Checklist:

- [x] Pull request title reflects its content
- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# Description:

Added the missing documentation and linked it in other documents where applicable.

~P.S.: only the last commit relevant (`.NET Bindings - Created quick_start_setting_up_csharp.adoc`) as it is based on PR #542~

### CODE REVIEW

File: docs/Antora/modules/getting_started/pages/quick_start_setting_up_csharp.adoc [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+For the cross-compilation we use a `bash` prompt on Linux(TM) (or WSL on Windows(TM)) to run the result.
+
+To start, create a `csharp/quick_start` directory and navigate to it.
+
+With  the following commands we first create an empty (.NET8.0) console application project and then we add the _openDAQ.Net_ Bindings NuGet package reference to the project, which contains all the needed binaries.
+
+[source,shell]
+----
+dotnet new console
+dotnet add package openDAQ.Net
```
Comment (aljazfrancic) [inline docs/Antora/modules/getting_started/pages/quick_start_setting_up_csharp.adoc:53]: This command fails for me with:
```
info : X.509 certificate chain validation will use the default trust store selected by .NET for code signing.
info : X.509 certificate chain validation will use the default trust store selected by .NET for timestamping.
info : Adding PackageReference for package 'openDAQ.Net' into project 'C:\Users\aljaz\csharp\csharp.csproj'.
info :   GET https://api.nuget.org/v3/registration5-gz-semver2/opendaq.net/index.json
info :   NotFound https://api.nuget.org/v3/registration5-gz-semver2/opendaq.net/index.json 129ms
info :   GET https://api.nuget.org/v3/registration5-gz-semver2/opendaq.net/index.json
info :   NotFound https://api.nuget.org/v3/registration5-gz-semver2/opendaq.net/index.json 193ms
error: There are no versions available for the package 'openDAQ.Net'.
```

**Review by aljazfrancic — COMMENTED** [review]
Look well written, however one of the commands appears to fail during testing.

File: docs/Antora/modules/getting_started/pages/quick_start_setting_up_csharp.adoc (hunk 1 above)
Comment (aljazfrancic) [reply docs/Antora/modules/getting_started/pages/quick_start_setting_up_csharp.adoc:53]: The command you provided gives:
```
Option '--source' expects a single argument but 2 were provided.
```
However, this worked:
```
dotnet add package openDAQ.Net --source path/to/folder
```

I would write the guide as follows:
With the following command we first create an empty (.NET8.0) console application project:
```
dotnet new console
```
Then we add the openDAQ.Net Bindings NuGet package reference to the project, which contains all the needed binaries:
```
dotnet add package openDAQ.Net
```
Alternatively, the latest versions of the C# bindings are available at https://docs-dev.opendaq.com/ and can be used via:
```
dotnet add package openDAQ.Net --source path/to/folder
```

File: docs/Antora/modules/getting_started/pages/quick_start_setting_up_csharp.adoc [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+Linux::
++
+--
+Cross-compiling on Windows(TM) and running the result on Linux(TM) using WSL:
+[source,shell]
+----
+dotnet build --runtime linux-x64
+wsl
+cd ./bin/Debug/net8.0/linux-x64/
+./quick_start
```
Comment (aljazfrancic) [inline docs/Antora/modules/getting_started/pages/quick_start_setting_up_csharp.adoc:101]: I had to do:
`sudo apt-get install -y dotnet-runtime-8.0`
before I could run `./quick_start` using WSL.
Then everything worked fine.
I'd amend the guide with this info.

Comment (Daniel-BB) [reply docs/Antora/modules/getting_started/pages/quick_start_setting_up_csharp.adoc:108]: This is (implicitly) mentioned in the requirements section. I reworked this a bit as well, but it still only refers generally to the installation of the .NET SDK or runtime on Linux/WSL.

File: docs/Antora/modules/getting_started/pages/quick_start_setting_up_csharp.adoc [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+exit
+----
+
+NOTE: `dotnet build --runtime linux-x64` builds the console application for Linux(TM) and uses the associated binaries from the NuGet package. The result (`./bin/Debug/net8.0/linux-x64/quick_start`) can then be run in WSL under Ubuntu 22.04 (see Requirements above).
+--
+====
+
+With the `run` command the project is being built (when there was a change after the last run) and the resulting `quick_start.exe` executed. +
```
Comment (aljazfrancic) [inline docs/Antora/modules/getting_started/pages/quick_start_setting_up_csharp.adoc:119]: Delete the `+` at the end of line.

**Review by aljazfrancic — APPROVED** [review]
Can be merged after a small typo is fixed.

File: docs/Antora/modules/getting_started/pages/quick_start_setting_up_csharp.adoc (hunk 3 above)
Comment (Daniel-BB) [reply docs/Antora/modules/getting_started/pages/quick_start_setting_up_csharp.adoc:119]: no, I want a line-break here but not to start a new paragraph ;)

**Pull request merged**